src/app/view/editor-main-view/data-views/connections.view.ts

```diff
@@ -292,9 +292,12 @@ export class ConnectionsView {
   }
 
   displayConnections(inputConnections: Connection[]) {
-    const connections = inputConnections.filter((c) =>
-      this.filterConnectionsToDisplay(c) &&
-      this.editorView.doCullCheckPositionsInViewport(c.getPath())
+    const connections = inputConnections.filter((c) => {
+        if (this.editorView.doCullCheckPositionsInViewport(c.getPath())) {
+          return this.filterConnectionsToDisplay(c);
+        }
+        return false;
+      }
```

Comment:
If `doCullCheckPositionsInViewport` is a good discriminant, and takes less performance than `filterConnectionsToDisplay`, I'll suggest to implement it that way. It takes less code, and dismiss early cases the way you want. Then `&&` operator ensures that the second operation is not called if the first returns `false` anyway.

```suggestion
const connections = inputConnections.filter((c) =>
      this.editorView.doCullCheckPositionsInViewport(c.getPath()) &&
      this.filterConnectionsToDisplay(c)
```

And the same for the other objects